src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseEndpoint.java

```diff
@@ -103,6 +103,27 @@ public CaseEndpoint(
     return ResponseEntity.ok(buildDetailedCaseDTO(caseObj, caseevents, iac));
   }
 
+  @RequestMapping(value = "/sampleunitids", method = RequestMethod.GET)
+  public ResponseEntity<List<CaseDetailsDTO>> findCasesBySampleUnitIds(
+      @RequestParam(value = "sampleUnitId") final List<UUID> sampleUnitIds,
+      @RequestParam(value = "caseevents", required = false) final boolean caseevents,
+      @RequestParam(value = "iac", required = false) final boolean iac) {
+
+    List<CaseDetailsDTO> cases =
+        sampleUnitIds
+            .stream()
+            .filter(e -> caseService.findCaseBySampleUnitId(e) != null)
```

Comment:
If we do a map here followed by the filter we only need to do the DB lookup once